java-client/src/main/java/org/apache/pegasus/client/PegasusAdminClient.java

```diff
@@ -230,4 +232,34 @@ public void dropApp(String appName, int reserveSeconds) throws PException {
           String.format("Drop app:%s failed! error: %s.", appName, error.toString()));
     }
   }
+
+  @Override
+  public void listApps(boolean onlyGetAvailableApps, List<app_info> appInfoList) throws PException {
+    if (!appInfoList.isEmpty()) {
+      throw new PException(
+          new IllegalArgumentException(
+              String.format("listApps failed: output parameters 'appInfoList' not empty.")));
+    }
+
+    configuration_list_apps_request request = new configuration_list_apps_request();
+    request.setStatus(app_status.AS_AVAILABLE);
+    if (!onlyGetAvailableApps) {
+      request.setStatus(app_status.AS_INVALID);
+    }
+
+    list_apps_operator app_operator = new list_apps_operator(request);
+    error_code.error_types error = this.meta.operate(app_operator, META_RETRY_MIN_COUNT);
+
+    if (error != error_code.error_types.ERR_OK) {
+      throw new PException(
+          String.format(
+              "List apps failed, query status:%s, error:%s.",
+              request.getStatus(), error.toString()));
+    }
+
+    configuration_list_apps_response response = app_operator.get_response();
+    for (int i = 0; i < response.infos.size(); i++) {
```

Comment:
The type of `response.infos` is ArrayList, why not directly assign it to `appInfoList`?